### PR TITLE
convolutional.cfg added with test data. Relating PR #16358

### PR DESCRIPTION
**Pull request by ashishkrshrivastava**
This PR adds 2-layer convolutional model with test data.

### CODE REVIEW

File: testdata/dnn/darknet/convolutional.cfg [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+height=4
+width=4
+
+#for SAME padding set pad=0, for manual padding set padding to desired value.
+[convolutional]
+filters=4
+size=3
+stride=2
+pad=1
+activation=leaky
```
Comment (dkurt) [inline testdata/dnn/darknet/convolutional.cfg]: Can you also add a layer with `padding = 1` from https://github.com/opencv/opencv_extra/pull/697/ after it?
like
```
#Convolution layer
[convolutional]
filters=5
size=3
stride=1
padding=1
activation=leaky
```

**Comment by alalek** [comment]
> darknet/convolutional.cfg_in.npy 

Test expecting other names:
- `convolutional_in.npy`
- `convolutional_out.npy`

File: testdata/dnn/darknet/convolutional.cfg (hunk 1 above)
Comment (ashishkrshrivastava) [reply testdata/dnn/darknet/convolutional.cfg]: @dkurt Done!

Comment (ashishkrshrivastava) [reply testdata/dnn/darknet/convolutional.cfg]: @dkurt Done!

**Comment by dkurt** [comment]
Looks like this is first test with weighted network from Darknet. So we need to save `convolutional.weights` as well.

**Comment by ashishkrshrivastava** [comment]
> Looks like this is first test with weighted network from Darknet. So we need to save `convolutional.weights` as well.

@dkurt , convolutional.weights added.

**Comment by ashishkrshrivastava** [comment]
@alalek @dkurt , is any improvement needed ?

**Comment by alalek** [comment]
Builds with automatic tests should be green (see main PR). Patches with red builds can't be merged.

File: testdata/dnn/darknet/convolutional.cfg [hunk 2]
```diff
@@ -0,0 +1,21 @@
+[net]
+batch=1
+channels=3
+height=4
+width=4
```
Comment (alalek) [inline testdata/dnn/darknet/convolutional.cfg]: Consider using different values for `weight`/`height`.

Comment (dkurt) [reply testdata/dnn/darknet/convolutional.cfg]: /cc @ashishkrshrivastava

Comment (ashishkrshrivastava) [reply testdata/dnn/darknet/convolutional.cfg]: @alalek @dkurt, sorry for such a late response. I have changed width value.

Comment (ashishkrshrivastava) [reply testdata/dnn/darknet/convolutional.cfg]: @alalek , assuming you meant `width/height`

**Pull request merged**